source/common/router/config_impl.cc

```diff
@@ -720,6 +720,23 @@ void RouteEntryImplBase::finalizeRequestHeaders(Http::RequestHeaderMap& headers,
   }
 }
 
+const VirtualCluster*
+RouteEntryImplBase::virtualCluster(const Http::HeaderMap& headers,
+                                   const StreamInfo::StreamInfo& stream_info) const {
+  const VirtualCluster* virtual_cluster = vhost_.virtualClusterFromEntries(headers);
+  if (virtual_cluster != nullptr && !virtual_cluster->name().has_value()) {
+    // This means virtual cluster has been configured but there is no match in the downstream
+    // headers. Let us check if there is a match in virtual host headers.
```

Comment:
Is `virtual host headers` an existing concept? Is it just referring to `request_headers_to_add/remove`?

Reply:
it is just `request_headers_to_add/remove`